### ISSUE TITLE
False positive TypeDoesNotContainType after calling strtoupper on nullable.

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2125]
This is related to one of the annoying edge cases in PHP.
Depending on reflection, what C parameter parsing API is used (and what the strict_types setting is), passing null to an internal function may be coerced to string, or may throw a TypeError.

- https://github.com/tysonandre/php-src-ast-analysis is a WIP to extract the param/return type information from php-src. Right now, param parsing is too incomplete to be used.

I assume psalm is inferring that the PossiblyNullArgument would be the cause of a TypeError, or would be fixed/handled elsewhere instead of suppressed.

Note that emitting TypeDoesNotContainType would be correct if strict_types was on.

```php
<?php

class MyClass {
    public function getStr() : ?string {
        return rand(0,1) ? 'test' : null;
    }
    /** @psalm-suppress PossiblyNullArgument */
    public static function test(MyClass $i) : void {
        $g = $i->getStr();
        $x = strtoupper($g);  // this is what gets suppressed
        // Observed: TypeDoesNotContainType - src/user.php:11:38 - string is always defined and non-null
        // Expected: No issue, but only when strict_types is not on.
        $c = 'prefix ' . (strtoupper($g ?? '') === 'x' ? 'xa' : 'ya');
        echo "$x, $c\n";
    }
}
MyClass::test(new MyClass());
```
